clang/lib/AST/PreorderAST.cpp

```diff
@@ -115,9 +168,23 @@ void PreorderAST::Sort(Node *N) {
   [&](const Node *N1, const Node *N2) {
 
     if (const auto *L1 = dyn_cast<LeafExprNode>(N1)) {
-      if (const auto *L2 = dyn_cast<LeafExprNode>(N2))
+      if (const auto *L2 = dyn_cast<LeafExprNode>(N2)) {
```

Comment:
This comparison function seems important enough that it should be a named function with comments describing what it does, not just a lambda function.

Reply:
I have moved the comparison code to a comparator function but I cannot move it outside of the Sort function because I need the lambda to capture the context to refer to the Ctx and Lex objects in the comparator.